### PR TITLE
Fix ERR_print_errors() format to match documentation.

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
Fixes #11743

**Comment by levitte** [comment]
Side note: I've been pondering refactoring all possible variants of hexdump output into a BIO filter...  that's an effort in itself and won't happen now, but might take away the growth of a gazillion different hexdump functions.

(no, I do not intend to block this PR...  if nothing else, it's a necessary temporary measure)

File: crypto/o_str.c [hunk 1]
```diff
@@ -143,26 +143,24 @@ int OPENSSL_hexstr2buf_ex(unsigned char *buf, size_t buf_n, size_t *buflen,
 
     for (p = (const unsigned char *)str, q = buf, cnt = 0; *p; ) {
         ch = *p++;
-        if (ch == ':')
+        /* A seperator of CH_ZERO means there is no seperator */
```
Comment (t-j-h) [inline crypto/o_str.c:146]: separator not seperator

**Comment by t8m** [comment]
The actual test source seems to be missing? Also do I get it right that the added hexstr2buf variant is solely for the test purposes?

**Comment by levitte** [comment]
> Also do I get it right that the added hexstr2buf variant is solely for the test purposes?

... or maybe it's for feature completeness (I've complained about the lack of feature completeness, so I can understand if efforts are being made)

File: crypto/o_str.c (hunk 1 above)
Comment (slontis) [reply crypto/o_str.c:146]: > Side note: I've been pondering refactoring all possible variants of hexdump output into a BIO

In that case it sounds like I should just make these functions internal for now.

**Comment by slontis** [comment]
> Also do I get it right that the added hexstr2buf variant is solely for the test purposes?

It is not really a one way function, so if a separator is added it should work both ways.

File: doc/man3/OPENSSL_hexchar2int.pod [hunk 2]
```diff
@@ -19,6 +19,7 @@ OPENSSL_buf2hexstr_ex, OPENSSL_buf2hexstr
                            const unsigned char *buf, long buflen);
  char *OPENSSL_buf2hexstr(const unsigned char *buf, long buflen);
 
+
```
Comment (levitte) [inline doc/man3/OPENSSL_hexchar2int.pod:22]: Remove this extra empty line

File: doc/man3/OPENSSL_hexchar2int.pod [hunk 3]
```diff
@@ -59,12 +60,12 @@ or -1 on error.
...
 
-OPENSSL_buf2hexstr_ex() and OPENSSL_hexstr2buf_ex() return 1 on
-success, or 0 on error.
+OPENSSL_buf2hexstr_ex() and OPENSSL_hexstr2buf_ex()
+return 1 on success, or 0 on error.
 
 =head1 COPYRIGHT
 
-Copyright 2016-2019 The OpenSSL Project Authors. All Rights Reserved.
+Copyright 2016-2020 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (levitte) [inline doc/man3/OPENSSL_hexchar2int.pod:68]: Actually, you can simply undo all changes in this file.

File: include/crypto/err.h [hunk 4]
```diff
@@ -15,4 +15,7 @@ void err_cleanup(void);
 int err_shelve_state(void **);
 void err_unshelve_state(void *);
 
+char *openssl_buf2hexstr_sep(const unsigned char *buf, long buflen, char sep);
+unsigned char *openssl_hexstr2buf_sep(const char *str, long *buflen,
+                                      const char sep);
```
Comment (levitte) [inline include/crypto/err.h:20]: This is wrongly placed, as these are not ERR functions per se.  Please move these declarations to `cryptlib.h` (side note: that header should never have been moved to `include/internal`, as it was meant as declarations of libcrypto private functions)

File: crypto/err/err_prn.c [hunk 5]
```diff
@@ -15,6 +15,7 @@
 #include <openssl/crypto.h>
 #include <openssl/buffer.h>
 #include <openssl/err.h>
+#include "crypto/err.h"
```
Comment (levitte) [inline crypto/err/err_prn.c:18]: Should not be needed at this point

File: crypto/o_str.c [hunk 6]
```diff
@@ -10,8 +10,12 @@
 #include "e_os.h"
 #include <limits.h>
 #include <openssl/crypto.h>
+#include "crypto/err.h"
```
Comment (levitte) [inline crypto/o_str.c:13]: Should not be needed

File: doc/man3/OPENSSL_hexchar2int.pod [hunk 7]
```diff
@@ -59,8 +59,8 @@ or -1 on error.
 OPENSSL_buf2hexstr() and OPENSSL_hexstr2buf()
 return a pointer to allocated memory, or NULL on error.
 
-OPENSSL_buf2hexstr_ex() and OPENSSL_hexstr2buf_ex() return 1 on
-success, or 0 on error.
+OPENSSL_buf2hexstr_ex() and OPENSSL_hexstr2buf_ex() return 1 on success,
+or 0 on error.
```
Comment (levitte) [inline doc/man3/OPENSSL_hexchar2int.pod:63]: If you undo this too, this file can drop entirely from your commit list,

File: crypto/err/err_prn.c (hunk 5 above)
Comment (slontis) [reply crypto/err/err_prn.c]: These last 3 were already addressed :)

**Comment by slontis** [comment]
I am just removing the err include from the test also..

**Comment by levitte** [comment]
My approval stands

**Comment by slontis** [comment]
Thanks. Merged to master.

**Pull request closed without merging**